docs/source/traits_user_manual/defining.rst

```diff
@@ -472,6 +472,65 @@ will raise a TraitError if a value of any other type is assigned. For example::
     >>> steven.manager_name = 5
     traits.trait_errors.TraitError: The 'manager_name' trait of an Employee instance must be a string or None, but a value of 5 <type 'int'> was specified.
 
+.. index:: Union trait
+
+.. _union:
+
+Union
+::::::
+The Union trait accepts a value that is considered valid by at least one
+of the traits in its definition. It is a simpler and therefore less error-prone
+alternative to the `Either` trait, which allows more complex constructs and
+may sometimes exhibit mysterious validation behaviour. The Union trait however,
+validates the value assigned to it against each of the traits in its definition
+in the order they are defined. Union only accepts trait types or trait type
+instances or None in its definition. Prefer to use Union over `Either` to
+remain future proof.
+
+.. index::
+   pair: Union trait; examples
+
+The following is an example of using Union::
+
+    # union.py --- Example of Union predefined trait
+
+    from traits.api import HasTraits, Union, Int, Float, Instance
+
+    class Employee(HasTraits):
+        manager_name = Union(Str, None)
+        salary = Union(Instance(Int), Instance(Float))
```

Comment:
This does not seem correct; I would expect `Union(Int, Float)`.

Reply:
Yes, I wanted to demonstrate using an Instance of a user defined class as an argument to Union. Made the change.